xrdp/xrdp_login_wnd.c

```diff
@@ -193,13 +381,29 @@ static int APP_CC
 xrdp_wm_ok_clicked(struct xrdp_bitmap *wnd)
 {
     struct xrdp_bitmap *combo;
+
+#ifndef XRDP_LOGINFIRST
     struct xrdp_bitmap *label;
     struct xrdp_bitmap *edit;
+#else
+    struct list *usr;
+    struct list *pw;
+#endif
+
     struct xrdp_wm *wm;
     struct xrdp_mod_data *mod_data;
+
+#ifndef XRDP_LOGINFIRST
     int i;
+#endif
 
     wm = wnd->wm;
+
+#ifdef XRDP_LOGINFIRST
+    usr = wm->mm->login_names;
```

Comment:
You can combine declaration and initialization to reduce the number of preprocessor directives and improve readability.